plugins/airtable/src/data.ts

```diff
@@ -372,3 +469,19 @@ export async function syncExistingCollection(
         return { didSync: false }
     }
 }
+
+function isBarcodeValue(value: unknown): value is { text: string } {
```

Comment:
Instead of doing `is { text: string }` can we guard that this is `value is BarcodeValue` and `value is AiTextValue` from the types in `api.ts`?